libsolidity/formal/SMTChecker.cpp

```diff
@@ -332,19 +331,18 @@ void SMTChecker::endVisit(VariableDeclarationStatement const& _varDecl)
 	{
 		if (auto init = _varDecl.initialValue())
 		{
-			auto symbTuple = dynamic_pointer_cast<smt::SymbolicTupleVariable>(m_expressions[init]);
-			/// symbTuple == nullptr if it is the return of a non-inlined function call.
-			if (symbTuple)
-			{
-				auto const& components = symbTuple->components();
-				auto const& declarations = _varDecl.declarations();
+			auto symbTuple = dynamic_pointer_cast<smt::SymbolicTupleVariable>(m_context.expression(*init));
+			solAssert(symbTuple, "");
```

Comment:
Now the expression of a non-inlined function call is created. If it has more than one value, it is a tuple without components.